### ISSUE TITLE
Model is not selecting correct shard_group

### ISSUE THREAD

**Issue by irttboopathy** [JBKahn/django-sharding#65]
When the model_config is configured with @model_config(shard_group='default', sharded_by_field='model_id') router is always sending None from method `db_for_read`. It uses hints to get the shard, but its always `{exact_lookups = {}}`

Please help resolving the issue.

Right now Model am using is, 

`@model_config(database='default')
class ShardedCarIDs(TableStrategyModel):
    pass

@model_config(shard_group='default')
class Car(models.Model):
	another_id = TableShardedIDField(primary_key=True, source_table_name=ShardedCarIDs)
	ignition_type = models.CharField(max_length=120)

	def get_shard(self):
		print 'get shard method'
		return 'app_shard_001'
		# from django.contrib.auth import get_user_model
		# return get_user_model().objects.get(pk=self.user_pk).shard

	@staticmethod
	def get_shard_from_id(user_pk):
		return 'app_shard_001'`

This is just for example, returning the shard db name.

**Comment by JBKahn** [comment]
What's an example of a call causing this?

**Comment by JBKahn** [comment]
I also updated your question formatting for it was a bit easier for me to read. Let me know if I fudged anything.

**Comment by irttboopathy** [comment]
Am calling `Car.objects.all()`

I might be wrong somewhere. In example `@model_config(sharded=True)` is not working.. Can you send links some working example so that it would be helpful for me.

**Comment by JBKahn** [comment]
You can see some working examples in the test code in this repository. You can't use `Car.objects.all()` because that would require Django to call each of your databases to get all the cars, something that is not done for you. You have to tell it which database you want to read from in this case, i.e. `Car.objects.using(first_database_name).all()`.

**Comment by irttboopathy** [comment]
Yeah.. I thought model's method `get_shard` before querying automatically in the framework. Anyways this works..

But IMHO if this method is called automatically from the framework, it would be too good.. 

Thanks.. 👍

**Comment by JBKahn** [comment]
It would have to know what you wanted to do to do that step for you. Or automatically always look into all the shards. The package is meant to give you application level sharding, putting that control in your hands. Unlike something more akin to https://www.citusdata.com/ where is transparent to your application and you'd treat like a single database.
